passageflex/src/main/java/id/passage/passageflex/PassageFlex.kt

```diff
@@ -0,0 +1,139 @@
+package id.passage.passageflex
+
+import android.app.Activity
+import id.passage.android.passageflex.api.AuthenticateAPI
+import id.passage.android.passageflex.api.RegisterAPI
+import id.passage.android.passageflex.model.AuthenticateWebAuthnFinishWithTransactionRequest
+import id.passage.android.passageflex.model.AuthenticateWebAuthnStartWithTransactionRequest
+import id.passage.android.passageflex.model.AuthenticatorAttachment
+import id.passage.android.passageflex.model.RegisterWebAuthnFinishWithTransactionRequest
+import id.passage.android.passageflex.model.RegisterWebAuthnStartWithTransactionRequest
+
+/**
+ *  Use `PassageFlex` to easily add passkeys to your existing authentication system on Apple devices.
+ *
+ *  Find out more at: https://passage.1password.com
+ */
+public object PassageFlex {
+    /**
+     * The base struct for utilizing Apple's native passkey APIs and Passage Flex APIs together.
+     */
+    public object Passkey {
+        /**
+         * Registers a new passkey.
+         *
+         * Prompts the user associated with the provided Passage `transactionId` to create and
+         * register a new passkey for use with your app.
+         *
+         * @param transactionId The Passage transaction id provided by your app's server.
+         * @param activity Activity to surface the Credentials Manager prompt within.
+         * @param authenticatorAttachment (Optional) The type of authentication that will be used in
+         * this WebAuthN flow request. Defaults to `AuthenticatorAttachment.platform`. Use
+         * `AuthenticatorAttachment.cross-platform` for physical security key registration.
+         * @return A single-use "nonce" from Passage server to be exchanged for an authentication
+         * token on your app's server.
+         * @throws RegisterException
+         */
+        public suspend fun register(
```

Comment:
Probably a nitpick but could/should the PassageFlex class be separated/broken down further? Would this help Android devs looking at the source code we expose in any way, or help make developing in Android flex any easier? In the JS PassageFlex I've kept the main class pretty barebones as an entrypoint and the passkey itself is a class, so are the authentication/register classes. It made sense in JS but not sure it makes sense here.

Just a thought and curious to hear yours!

Reply:
Thanks for the recommendation! I have broken it out further, matching the iOS structure.